### PR TITLE
OPEN: Minor CI and Readme Improvements

### PR DESCRIPTION
**Pull request by Victor-Jung**
In this PR, we:
- Improves the README, explaining how to get started with Deeploy using the Docker container. In the future I will add more content such as 'How to install dependencies manually', 'How to add tests to the CI', and 'How to add new dependencies'.
- Update the link of the docker container used to run the CI; previously, it was pointing at my fork.
- BUGFIX: Add a step to retry on timeout for the tests of large networks on Siracusa.
- Schedule a CI run every 6 days at 2am to keep the cache fresh.

### CODE REVIEW

**Review by Xeratec — APPROVED** [review]
LGTM. 
Do you want to add a section explaining the CI in the Readme in this PR or another one? As this PR is not a critical change I suggest adding it here. Otherwise feel free to merge.

**Comment by Victor-Jung** [comment]
@Xeratec I improved the README and added a read `Getting Started` section for beginners. More detailed tutorials on the CI and CD flow will come later if that's fine for you.

File: .github/workflows/TestRunnerTiledSiracusa.yml [hunk 1]
```diff
@@ -47,11 +47,16 @@ jobs:
...
           key: ${{ runner.os }}-ccache
       - name: Run Test
-        run: |
-          cd DeeployTest
-          mkdir -p /app/.ccache
-          export CCACHE_DIR=/app/.ccache
-          source /app/install/pulp-sdk/configs/siracusa.sh
-          python testRunner_tiled_siracusa.py -t Tests/${{ inputs.test-name }} --cores=${{ inputs.num-cores }} --l1 ${{ matrix.L1 }} --defaultMemLevel=${{ inputs.default-memory-level }} ${{ inputs.double-buffer && '--doublebuffer' || '' }}
-        shell: bash
+        uses: nick-fields/retry@v3
```
Comment (Xeratec) [inline .github/workflows/TestRunnerTiledSiracusa.yml:50]: Do we have more insight into why this test times out sometimes?

File: .github/workflows/TestRunnerTiledSiracusaWithNeureka.yml [hunk 2]
```diff
@@ -51,11 +51,16 @@ jobs:
...
+          timeout_minutes: 5
+          max_attempts: 3
+          retry_on: timeout
+          command: |
+            cd DeeployTest
+            mkdir -p /app/.ccache
+            export CCACHE_DIR=/app/.ccache
+            source /app/install/pulp-sdk/configs/siracusa.sh
+            python testRunner_tiled_siracusa_w_neureka.py -t Tests/${{ inputs.test-name }} --cores=${{ inputs.num-cores }} --l1 ${{ matrix.L1 }} --defaultMemLevel=${{ inputs.default-memory-level }} ${{ inputs.double-buffer && '--doublebuffer' || '' }} ${{ inputs.neureka-wmem && '--neureka-wmem' || '' }}
+          shell: bash
```
Comment (Xeratec) [inline .github/workflows/TestRunnerTiledSiracusaWithNeureka.yml:65]: Same here, do we have more insight into why this test times out sometimes?

**Review by Xeratec — APPROVED** [review]
LGTM, feel free to merge. 
One question, tough. Do we have more insight into why some test on Siracusa time out sometimes? If it is a problem in our codebase, please open an issue.

File: .github/workflows/TestRunnerTiledSiracusa.yml (hunk 1 above)
Comment (Xeratec) [reply .github/workflows/TestRunnerTiledSiracusa.yml:50]: But at which point is it happening? When running Deeploy or during the simulation with GVSoC?

Comment (Victor-Jung) [reply .github/workflows/TestRunnerTiledSiracusa.yml:50]: It happens during compilation :/

Comment (Xeratec) [reply .github/workflows/TestRunnerTiledSiracusa.yml:50]: Okay, then I suggest you open an issue, reference this conversation/changes as a temporary bug fix in the issue, and merge this PR.

Comment (Victor-Jung) [reply .github/workflows/TestRunnerTiledSiracusa.yml:50]: Sounds resonable

**Pull request merged**